OpenRA.Mods.Common/Widgets/WidgetUtils.cs

```diff
@@ -419,4 +419,40 @@ public U Update(T input)
 			return lastOutput;
 		}
 	}
+
+	public class PredictedCachedTransform<T, U>
+	{
+		readonly Func<T, U> transform;
+
+		bool initialized;
+		T lastInput;
+		U lastOutput;
+
+		bool predicted;
+		U prediction;
+
+		public PredictedCachedTransform(Func<T, U> transform)
+		{
+			this.transform = transform;
+		}
+
+		public void Predict(U value)
+		{
+			predicted = true;
+			prediction = value;
+		}
+
+		public U Update(T input)
+		{
+			if ((predicted || initialized) && ((input == null && lastInput == null) || (input != null && input.Equals(lastInput))))
```

Comment:
What happens if it predicts the value and then the value never changes? It would now effectively be showing the incorrect value.

Reply:
We could add a sanity check and subscribe it to LobbyOptionsChanged event

Reply:
`input` here is `orderManager.LobbyInfo.GlobalSettings` (not the lobby option itself!) which is what also triggers the `LobbyOptionsChanged` event 😉

Reply:
oh I guess LobbyOptionsChanged event still wouldn't work as it won't get called when the change has been refused

Reply:
They all will update whenever the server sends new `LobbyInfo` data, the response to this specific option change doesn't specifically matter.

Reply:
It will happen often enough for people to notice and probably file a bug report. But I guess it's acceptable

Reply:
That sounds like we'd generate a less severe kind of #5802.